File: exam-system-java/entity/src/main/java/com/yyjy/exam/entity/exam/entity/ExamRecords.java
package com.yyjy.exam.entity.exam.entity;

import com.yyjy.exam.entity.paper.entity.Paper;
import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDateTime;
import java.util.List;

@Entity
public interface ExamRecords {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int id();

    @ManyToOne
    Paper paper();

    @IdView
    int paperId();

    String studentName();

    @Nullable
    Integer score();

    @Nullable
    String answers();

    @Nullable
    LocalDateTime startTime();

    @Nullable
    LocalDateTime endTime();

    @Nullable
    ExamRecordStatus status();

    @Nullable
    Integer windowSwitches();

    @OneToMany(mappedBy = "examRecord")
    List<AnswerRecord> answerRecords();

    @Nullable
    LocalDateTime createTime();

    @Nullable
    LocalDateTime updateTime();

    @Column(name = "deleted_at")
    @LogicalDeleted("now")
    @Nullable
    LocalDateTime delFlag();
}

File: exam-system-server/src/main/kotlin/com/yyjy/models/entity/ExamRecords.kt
package com.yyjy.models.entity

import org.babyfish.jimmer.sql.*
import java.time.LocalDateTime

/**
 * 考试记录表（考试过程和结果）
 */
@Entity
interface ExamRecords {

    @Id
    @GeneratedValue(
        strategy = GenerationType.IDENTITY
    )
    val id: Int

    val examId: Int

    val studentName: String

    val score: Int?

    /**
     * 答案
     */
    val answers: String?

    val startTime: LocalDateTime?

    val endTime: LocalDateTime?

    val status: String?

    /**
     * 窗口切换
     */
    val windowSwitches: Int?

    /**
     * 创建时间
     */
    val createTime: LocalDateTime?

    /**
     * 更新时间
     */
    val updateTime: LocalDateTime?

    /**
     * 删除时间
     */
    @Column(name = "deleted_at")
    @LogicalDeleted("now")
    val delFlag: LocalDateTime?
}

